lib/AST/NameLookup.cpp

```diff
@@ -1253,38 +1253,15 @@ void ClassDecl::recordObjCMethod(AbstractFunctionDecl *method) {
   vec.push_back(method);
 }
 
-static bool checkAccessibility(const DeclContext *useDC,
-                               const DeclContext *sourceDC,
-                               Accessibility access) {
-  if (!useDC)
+bool ValueDecl::isAccessibleFrom(const DeclContext *DC) const {
+  auto access = getFormalAccess(DC);
+  if (!DC)
```

Comment:
Yea, this confuses me as well, but this behavior is documented and a number of things break without it. I can dig in a bit more as to why.